common/service/config/config.go

```diff
@@ -69,6 +69,8 @@ type (
 		Port int `yaml:"port"`
 		// BindOnLocalHost is true if localhost is the bind address
 		BindOnLocalHost bool `yaml:"bindOnLocalHost"`
+		// BindOnLocalHost is true if localhost is the bind address
+		BindOnAllInterfaces bool `yaml:"bindOnAllInterfaces"`
```

Comment:
Instead, I would suggest adding a parameter `BindOnIP string`.
And then parse it with `net.ParseIP(BindOnIP)`.
And yes we need to leave `BindOnLocalHost` for backwards compatibility.

Reply:
True, waiting for cadence team to suggest what they think is best for them

Reply:
artms - I prefer BindOnIP to BindOnAllInterfaces. It gives maximum flexibility i.e. we no longer need special options for BindOnAll or BindOnLocalHost

Reply:
Fair enough, what about backwards compatibility, should I fully eradicated BindOnLocalHost or make something like:
1. if `BindOnLocalhost` is set and BindOnIP is set at the same time - "crash" with error that values are mutually exclusive
2. if `BindOnIP` is not set - old behavior where we try listening on external machine IP
3. `BindOnIP` is set - do as setting specifies

Reply:
We have a few internal development configs that depend on BindOnLocaHost. So, lets leave it aside for backwards compatibility for now. I created a tracking task to get to this later
https://github.com/uber/cadence/issues/1300

To sumarize, lets go with the following behavior
(1) If both BindOnIP and BindOnLocalHost are set - panic
(2) If BindOnLocalHost or BindOnIP is set, use it
(3) Else use default behavior